dbt/include/snowflake/macros/adapters.sql

```diff
@@ -134,20 +134,29 @@
 
 {% endmacro %}
 
-{% macro snowflake__list_relations_without_caching(schema_relation, max_iter=10, max_results_per_iter=10000) %}
+{% macro snowflake__list_relations_without_caching(schema_relation, max_iter=10, max_results_per_iter=10000, query_pre_hook=None) %}
+  {# -- Use query_pre_hook for optional configurations prior to fetching relations. For sake of
+     -- consistent use, place ;s in the query_pre_hook definition, not this macro. #}
 
   {%- set max_total_results = max_results_per_iter * max_iter -%}
   {%- set sql -%}
+    {% if query_pre_hook %}
```

Comment:
this is a conservative move because I think we could just not have the guard, but it's very easy to cause iterable unpack errors from empty queries (only `;`) and this helps avoid that.